lib/ethpy/ethpy/hyperdrive/api.py

```diff
@@ -259,6 +259,93 @@ def get_in_for_out(
+            The maturity timestamp of the open position, in epoch seconds.
+
+        Returns
+        -------
+        tuple[FixedPoint, FixedPoint, FixedPoint] consisting of:
+            curve_fee : FixedPoint
+                Curve fee, in shares.
+            flat_fee : FixedPoint
+                Flat fee, in shares.
+            gov_fee : FixedPoint
+                Governance fee, in shares.
+        """
+        if maturity_time is None:
+            maturity_time: int = self.pool_config["positionDuration"]
+        time_remaining_in_seconds = FixedPoint(maturity_time - self.current_block_time)
+        normalized_time_remaining = time_remaining_in_seconds / self.pool_config["positionDuration"]
+        curve_fee = (
+            (FixedPoint(1) - self.spot_price) * self.pool_config["curveFee"] * bonds_in * normalized_time_remaining
+        ) / self.pool_config["initialSharePrice"]
+        flat_fee = (
+            bonds_in * (FixedPoint(1) - normalized_time_remaining) * self.pool_config["flatFee"]
+        ) / self.pool_config["initialSharePrice"]
+        gov_fee = curve_fee * self.pool_config["governanceFee"] + flat_fee * self.pool_config["governanceFee"]
+        return curve_fee, flat_fee, gov_fee
+
+    def calculate_fees_out_given_shares_in(
+        self, shares_in: FixedPoint, maturity_time: int | None = None
+    ) -> tuple[FixedPoint, FixedPoint, FixedPoint]:
+        """Calculates the fees that go to the LPs and governance.
+
+        Implements the formula:
+            curve_fee = ((1 / p) - 1) * phi_curve * c * dz
+            gov_fee = shares * phi_gov
+            flat_fee = (d_y * (1 - t) * phi_flat) / c
+
+        Arguments
+        ---------
+        bonds_in : FixedPoint
+            The amount of bonds in.
+        interface : HyperdriveInterface
+            The API interface object.
+        maturity_time : int, optional
+            The maturity timestamp of the open position, in epoch seconds.
+
+        Returns
+        -------
+        tuple[FixedPoint, FixedPoint, FixedPoint] consisting of:
+            curve_fee : FixedPoint
+                Curve fee, in shares.
+            flat_fee : FixedPoint
+                Flat fee, in shares.
+            gov_fee : FixedPoint
+                Governance fee, in shares.
+        """
+        if maturity_time is None:
+            maturity_time: int = self.pool_config["positionDuration"]
+        time_remaining_in_seconds = FixedPoint(maturity_time - self.current_block_time)
+        normalized_time_remaining = time_remaining_in_seconds / self.pool_config["positionDuration"]
+        curve_fee = (
+            ((FixedPoint(1) / self.spot_price) - FixedPoint(1))
+            * self.pool_config["curveFee"]
+            * self.pool_config["initialSharePrice"]
+            * shares_in
+        )
+        flat_fee = (
+            shares_in * (FixedPoint(1) - normalized_time_remaining) * self.pool_config["flatFee"]
+        ) / self.pool_config["initialSharePrice"]
```

Comment:
We don't have this in our implementation because a flat fee is never applied when we use this (it's only used in `openLong` so we know the trade is 100% on the curve): https://github.com/delvtech/hyperdrive/blob/main/contracts/src/HyperdriveBase.sol#L333.

Reply:
Ok good to know. We decided to leave it in, just in case this gets used in a context outisde of `openLong`)